lib/gemstash/cli.rb

```diff
@@ -66,5 +67,20 @@ def status
     def stop
       Gemstash::CLI::Stop.new(self).run
     end
+
+    desc "preload", "Preloads all the gems in your gemstash server from the default upstream"
+    method_option :latest, :type => :boolean, :default => false, :desc =>
+      "Only fetch the latest specs"
+    method_option :server_url, :type => :string, :default => "http://localhost:9292", :desc =>
+      "Gemstash server url"
```

Comment:
No, it is not the upstream url, I do create an `Upstream` just to reuse the object with the client (https://github.com/bundler/gemstash/pull/48/files#diff-b6f92f551dd149bd7013ca6f7f0fd7a2R17)

What this does is create an HTTPClient and start requesting the *head* of the gems to gemstash without using any upstream path, so *it* downloads those from the default upstream. This process will not make any call to rubygems.org directly.

All this does is fake a client for every gem version there is in the specs file.